crates/walnut/src/verification.rs

```diff
@@ -0,0 +1,202 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use anyhow::{anyhow, Context, Result};
+use console::{pad_str, Alignment, Style, StyledObject};
+use dojo_world::metadata::get_default_namespace_from_ws;
+use dojo_world::migration::strategy::MigrationStrategy;
+use futures::future::join_all;
+use reqwest::StatusCode;
+use scarb::core::Workspace;
+use serde::Serialize;
+use serde_json::Value;
+use walkdir::WalkDir;
+
+/// Verifies all classes declared during migration.
+/// Only supported on hosted networks (non-localhost).
+pub async fn walnut_verify_migration_strategy(
+    ws: &Workspace<'_>,
+    rpc_url: String,
+    migration_strategy: &MigrationStrategy,
+) -> Result<()> {
+    let ui = ws.config().ui();
+    // Check if rpc_url is localhost
+    if rpc_url.contains("localhost") || rpc_url.contains("127.0.0.1") {
+        ui.print(" ");
+        ui.print("Verifying classes with Walnut is only supported on hosted networks.");
+        ui.print(" ");
+        return Ok(());
+    }
+
+    // its path to a file so `parent` should never return `None`
+    let root_dir: &Path = ws.manifest_path().parent().unwrap().as_std_path();
+    let default_namespace = get_default_namespace_from_ws(ws)?;
+
+    // Check if there are any contracts or models in the strategy
+    if migration_strategy.contracts.is_empty() && migration_strategy.models.is_empty() {
+        ui.print(" ");
+        ui.print("🌰 No contracts or models to verify.");
+        ui.print(" ");
+        return Ok(());
+    }
+
+    // Notify start of verification
+    ui.print(" ");
+    ui.print("🌰 Verifying classes with Walnut...");
+    ui.print(" ");
+
+    // Retrieve the API key and URL from environment variables
+    let api_key = std::env::var("WALNUT_API_KEY").context("WALNUT_API_KEY not set")?;
+    let api_url =
+        std::env::var("WALNUT_API_URL").unwrap_or_else(|_| "https://api.walnut.dev".to_string());
```

Comment:
Have seen `api.walnut.dev` and `app.walnut.dev`, for maintainability and iterations, we may keep those in the `lib.rs` or other constant file to easily modify them on demand and be sure we don't miss any update.